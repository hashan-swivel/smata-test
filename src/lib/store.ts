// In your 'store.ts' or where you set up your Redux store
import { modalReducer } from '@/lib/features/modal/modalSlice';
import { combineReducers, configureStore } from '@reduxjs/toolkit';

const rootReducer = combineReducers({
  modal: modalReducer
});

export const store = configureStore({
  reducer: rootReducer
});

export type AppStore = typeof store;
export type AppState = ReturnType<AppStore['getState']>;
export type AppDispatch = AppStore['dispatch'];
