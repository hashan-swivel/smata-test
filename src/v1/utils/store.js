import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import counter from '../reducers';

const store = (initialState) =>
  createStore(counter, initialState, composeWithDevTools(applyMiddleware(thunk)));

export default store;
