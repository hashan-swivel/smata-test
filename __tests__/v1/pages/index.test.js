import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../src/utils/store';
import Home from '../../pages/index';

const mountComponent = () => {
  const mockStore = store({});

  return shallow(
    <Provider store={mockStore}>
      <Home />
    </Provider>
  );
};

describe('Home component', () => {
  it('should render', () => {
    const component = mountComponent();
    expect(component.length).toEqual(1);
  });

  // TODO: this test is breaking because of Tooltip library
  // Snapshot
  // it("renders correctly", () => {
  //   const tree = renderer.create(mountComponent()).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
