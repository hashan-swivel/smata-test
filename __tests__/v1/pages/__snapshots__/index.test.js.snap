// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
Array [
  <div
    className="alert"
  />,
  <header
    className="header"
  >
    <div
      className="user-wrapper"
    >
      <nav>
        <div
          className="nav-left"
        >
          <a
            className="logo light"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "backgroundImage": "",
              }
            }
            target="_self"
            title={null}
          >
            SMATA
          </a>
        </div>
        <div
          className="navbar-menu-logout"
        >
          <a
            className="icon icon-help-white"
            href="https://help.smata.com/hc/en-us"
            rel="nofollow noopener noreferrer"
            style={
              Object {
                "backgroundImage": "",
              }
            }
            target="_blank"
            title={null}
          >
            Help Centre
          </a>
          <a
            className="icon icon-user-white"
            href="undefined/login"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "backgroundImage": "",
              }
            }
            target="_self"
            title={null}
          >
            Login
          </a>
        </div>
      </nav>
    </div>
  </header>,
  <div
    className=""
  >
    <div
      className="container"
    >
      <section
        className="landing-page"
      >
        <div
          className="wrapper"
        >
          <img
            alt="Opps this page has gone missing"
            className="landing-page-image"
            src="test-file-stub"
          />
          <div
            className="landing-page-text"
          >
            <h2>
              We are building something special
            </h2>
            <a
              className="button"
              href="https://www.smata.com"
              rel="nofollow noopener noreferrer"
              style={
                Object {
                  "backgroundImage": "",
                }
              }
              target="_blank"
              title={null}
            >
              Go to smata.com
            </a>
          </div>
        </div>
      </section>
    </div>
    <footer
      className=""
    >
      <div
        className="wrapper"
      >
        <div
          className="footer-sitemap"
        >
          <span
            className="footer-sitemap-legal"
          >
            © 
            2020
             Smata Technologies Pty Ltd - 
            <a
              href="undefined/privacy_policy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy Policy
            </a>
             - 
            <a
              href="undefined/terms_and_conditions"
              rel="noopener noreferrer"
              target="_blank"
            >
              Terms & Conditions
            </a>
          </span>
          <div
            className="social-links"
          >
            <div
              className="social-links-nav"
            >
              <a
                aria-label="Facebook"
                href="https://www.facebook.com/smatastrata"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-facebook-square fa-w-14 fa-2x "
                  data-icon="facebook-square"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M400 32H48A48 48 0 0 0 0 80v352a48 48 0 0 0 48 48h137.25V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.27c-30.81 0-40.42 19.12-40.42 38.73V256h68.78l-11 71.69h-57.78V480H400a48 48 0 0 0 48-48V80a48 48 0 0 0-48-48z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </a>
              <a
                aria-label="Instagram"
                href="https://www.instagram.com/smatastrata/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-instagram fa-w-14 fa-2x "
                  data-icon="instagram"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>,
]
`;
